### PR TITLE
Launcher fixes2

### PR DESCRIPTION
**Pull request by DjWarmonger**
- Fixed resizing of toolbar: https://bugs.vcmi.eu/view.php?id=2728
- Fixed icon update when a sub-mod is enabled or disabled: https://bugs.vcmi.eu/view.php?id=3217

### CODE REVIEW

File: launcher/modManager/cmodlistview_moc.cpp [hunk 1]
```diff
@@ -66,14 +66,15 @@ void CModListView::setupModsView()
...
 
-	connect(filterModel, SIGNAL(modelReset()),
-		this, SLOT(modelReset()));
+	connect(filterModel, &CModFilterModel::modelReset, this, &CModListView::modelReset);
 
-	connect(modModel, SIGNAL(dataChanged(QModelIndex,QModelIndex)),
-		this, SLOT(dataChanged(QModelIndex,QModelIndex)));
+	connect(modModel, &CModListModel::dataChanged, this, &CModListView::dataChanged);
+
+	//Suprisingly model is not automagically connected to filter model
```
Comment (Nordsoft91) [inline launcher/modManager/cmodlistview_moc.cpp:76]: automagically - typo?

File: launcher/modManager/cmodlistview_moc.cpp [hunk 2]
```diff
@@ -66,14 +66,15 @@ void CModListView::setupModsView()
...
+	connect(modModel, &CModListModel::dataChanged, filterModel, &QAbstractItemModel::dataChanged);
```
Comment (Nordsoft91) [inline launcher/modManager/cmodlistview_moc.cpp:77]: does it fix https://github.com/vcmi/vcmi/issues/812?

**Review by kambala-decapitator — APPROVED** [review]
looks good to me

**Review by Nordsoft91 — CHANGES_REQUESTED** [review]
Bug discovered

File: launcher/modManager/cmodmanager.cpp [hunk 3]
```diff
@@ -232,10 +232,12 @@ bool CModManager::doEnableMod(QString mod, bool on)
 
 	modSettings = writeValue(path, modSettings, QVariant(on)).toMap();
 	modList->setModSettings(modSettings["activeMods"]);
-	modList->modChanged(mod);
 
 	JsonUtils::JsonToFile(settingsPath(), modSettings);
 
+	//model is read from filesystem once it is saved
+	modList->modChanged(mod);
```
[the remark this answers is not in the document]
Comment (Nordsoft91) [reply launcher/modManager/cmodmanager.cpp:239]: If you asking about bug https://github.com/vcmi/vcmi/pull/822#issuecomment-1236968223, it's a regression (new bug)

[the remark this answers is not in the document]
Comment (nullkiller) [reply launcher/modManager/cmodmanager.cpp:239]: Yes I meant it.

**Comment by nullkiller** [comment]
> Tested the build. It has major issue (regression):
> 
> 1. Press "download and refresh repositories"
> 2. Remove mod available on the server (VCMI extras for instance)
> 3. Observe problem
> 
> Expected result: enable button is not available, red circle button should not be visible Actual result: removed mod is possible to be enabled, pressing enable you will observe error

@DjWarmonger do you plan to fix this?

**Comment by nullkiller** [comment]
Please rebase on top of develop.

**Pull request closed without merging**